src/utils/daos_hdlr.c

```diff
@@ -2973,79 +2974,73 @@ dm_connect(struct cmd_args_s *ap,
 						     ca->dst_c_uuid,
 						     &attr, NULL, NULL);
 				if (rc != 0) {
-					fprintf(ap->errstream, "failed to "
-						"create destination "
-						"container: %d\n", rc);
-						D_GOTO(err_dst_root,
-						       rc);
+					fprintf(ap->errstream,
+						"failed to create destination container: %d %s\n",
+						rc, strerror(rc));
+					D_GOTO(err_dst_root, rc = daos_errno2der(rc));
 				}
 			} else {
 				rc = daos_cont_create(ca->dst_poh,
 						      ca->dst_c_uuid,
 						      props, NULL);
 				if (rc != 0) {
-					fprintf(ap->errstream, "failed to "
-						"create destination "
-						"container: %d\n", rc);
-						D_GOTO(err_dst_root,
-						       rc);
+					fprintf(ap->errstream,
+						"failed to create destination container: %d\n",
+						rc);
+					D_GOTO(err_dst_root, rc);
 				}
 			}
 			rc = daos_cont_open(ca->dst_poh, ca->dst_c_uuid,
 					    DAOS_COO_RW, &ca->dst_coh,
 					    dst_cont_info, NULL);
 			if (rc != 0) {
 				fprintf(ap->errstream,
-					"failed to open container: "
-					"%d\n", rc);
+					"failed to open container: %d\n", rc);
 				D_GOTO(err_dst_root, rc);
 			}
 			fprintf(ap->outstream,
 				"Successfully created container "
 				""DF_UUIDF"\n", DP_UUID(ca->dst_c_uuid));
 		} else if (rc != 0) {
-			fprintf(ap->errstream, "failed to open container: "
-				"%d\n", rc);
+			fprintf(ap->errstream, "failed to open container: %d\n", rc);
 			D_GOTO(err_dst_root, rc);
 		}
 		if (is_posix_copy) {
 			rc = dfs_mount(ca->dst_poh, ca->dst_coh, O_RDWR,
 				       &dst_file_dfs->dfs);
 			if (rc != 0) {
 				fprintf(ap->errstream,
-				"dfs mount on destination "
-				"failed: %d\n", rc);
-				D_GOTO(err_dst, rc);
+					"dfs mount on destination failed: %d\n", rc);
+				D_GOTO(err_dst, rc = daos_errno2der(rc));
 			}
 		}
 	}
 	D_GOTO(out, rc);
 err_dst:
-	rc = daos_cont_close(ca->dst_coh, NULL);
-	if (rc != 0) {
-		fprintf(ap->errstream, "failed to close destination "
-			"container (%d)\n", rc);
-	}
+	rc2 = daos_cont_close(ca->dst_coh, NULL);
```

Comment:
rc was being reset here so the function was not returning failure, leading to errors later on.